File: AuctionWebApp.Server/Data/Dto/CatalogRequest.cs
namespace AuctionWebApp.Server.Data.Dto
{
    public class CatalogRequest
    {
        public ulong? UserId { get; set; }
        public string SearchString { get; set; } = "";
        public ulong MinPrice { get; set; }
        public ulong? MaxPrice { get; set; }
        public byte SelectedSorter { get; set; }
        public short ItemsOnPage { get; set; }
        public int PageNumber { get; set; }
        public short CategoryId { get; set; }
        public Dictionary<byte, string>? AuctionTypes { get; set; }
        public bool[] TypeChecked { get; set; } = [];
        public Dictionary<byte, string>? Conditions { get; set; }
        public bool[] CondChecked { get; set; } = [];
    }
}

File: auctionwebapp.client/src/app/model/catalogRequest.ts
export class CatalogRequest {
  constructor(
    public userId?: number,
    public searchString: string = "",
    public minPrice: number = 0,
    public maxPrice?: number,
    public selectedSorter: number = 5,
    public itemsOnPage: number = 15,
    public pageNumber: number = 1,
    public categoryId: number = 0,
    public auctionTypes?: Map<number, string>,
    public typeChecked: Array<boolean> = new Array(),
    public conditions?: Map<number, string>,
    public condChecked: Array<boolean> = new Array()
  ) {}
}
